src/Entity/Form/SubscriptionForm.php

```diff
@@ -67,6 +68,16 @@ public function getFormId() {
     return parent::getFormId() . '_' . $this->entity->getRatePlan()->id();
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state) {
+    $form = parent::form($form, $form_state);
+    // Redirect to Rate Plan detail page on submit.
+    $form['#action'] = $this->getEntity()->getRatePlan()->url();
```

Comment:
I think this should be `$form['#action'] = $this->getEntity()->getRatePlan()->url('subscribe');` because the canonical view display could be configured to show a link to the subscribe form instead of the rendered form.